nft-client/src/main.rs

```diff
@@ -334,14 +370,55 @@ async fn main() -> anyhow::Result<()> {
                 .context("Failed to connect to Node GRPC")?;
 
             let consensus_status = grpc_client.get_consensus_status().await?;
-            let instance_info = grpc_client
-                .get_instance_info(contract.0, &consensus_status.best_block)
+            let contract_context =
+                smart_contracts::ContractContext::new(contract.0, smart_contracts::ReceiveName {
+                    name: String::from("CIS2-NFT.view"),
```

Comment:
It would be better to add a constructor to ReceiveName, rather than unfolding it like this. The fact we store a String internally should not be used.